### PR TITLE
feat(capture): make endpoint return 503 on produce errors

### PR DESCRIPTION
**Pull request by hazzadous**
At the moment we are swallowing errors. This change should make it such
that when e.g. we have issues with Kafka, the client is informed that
there is an issue, and we can also use HTTP error rates to monitor and
alert, rather than looking at Kafka specific metrics.

## Changes

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: posthog/api/capture.py [hunk 1]
```diff
@@ -289,6 +296,23 @@ def get_event(request):
...
+        try:
+            future.get(timeout=10)
+        except KafkaError:
+            # TODO: distinguish between retriable errors and non-retriable
+            # errors, and set Retry-After header accordingly.
+            return cors_response(
+                request,
+                generate_exception_response(
+                    "capture",
+                    "Unable to store event. Please try again. If you are the owner of this app you can check the logs for further details.",
```
Comment (yakkomajuri) [inline posthog/api/capture.py:309]: I guess events rather than event? Unfortunately it's possible that some go through and some don't but 🤷‍♂️

**Review by yakkomajuri — COMMENTED** [review]
I mean this looks sane? I read through the kafka-python docs and this makes sense but I just feel like there's something here I don't fully grasp?

If you're confident do merge but maybe we deploy this to dev first?

Also what impact will this have on our endpoint latency?

File: posthog/api/capture.py (hunk 1 above)
Comment (hazzadous) [reply posthog/api/capture.py:309]: Yeah we could give some feedback here on which messages in the batch failed, maybe worth it.

Comment (yakkomajuri) [reply posthog/api/capture.py:309]: nice to have, but not a blocker

**Comment by hazzadous** [comment]
> I mean this looks sane? I read through the kafka-python docs and this makes sense but I just feel like there's something here I don't fully grasp?

Mind explaining? Could be something I've not understood.

> If you're confident do merge but maybe we deploy this to dev first?

I've added an env to turn this off and off based on team id, so should be able to have at least a little control over rolling this out.

> Also what impact will this have on our endpoint latency?

It will almost certainly increase it. By how much I do not know yet.

**Comment by hazzadous** [comment]
I'll get this in and monitor and decide if it's feasible or not as it is.

**Comment by ellie** [comment]
omg yes I love this, thank you Harry 🙏

**Pull request merged**